pennylane/transforms/split_tape.py

```diff
@@ -264,6 +268,10 @@ def processing_fn(expanded_results):
         # sort results by idx
         results = tuple(res_dict[key] for key in sorted(res_dict))
 
+        if qml.math.requires_grad(expanded_results[0]):
+            # when computing gradients, we need to convert the tuple to a gradient box
+            return qml.math.stack(results)
```

Comment:
The stacking should be done on the user side for the new return type, I think you just need to add `and not qml.active_return()`